### PR TITLE
Fix issue with Podfile parsing. Add tests.

### LINKED ISSUE #546: Issue in Podfile parser
```
java.lang.ArrayIndexOutOfBoundsException: Index 1 out of bounds for length 1
        at com.defold.extender.services.CocoaPodsService.createMainPodfile(CocoaPodsService.java:447)
        at com.defold.extender.services.CocoaPodsService.resolveDependencies(CocoaPodsService.java:1381)
        at com.defold.extender.Extender.resolve(Extender.java:2619)
        at com.defold.extender.AsyncBuilder.asyncBuildEngine(AsyncBuilder.java:112)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:355)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
        at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
        at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:768)
        at org.springframework.aop.interceptor.AsyncExecutionInterceptor.lambda$invoke$0(AsyncExecutionInterceptor.java:113)
        at io.micrometer.context.ContextSnapshot.lambda$wrap$1(ContextSnapshot.java:106)
        at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
        at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:304)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
        at java.base/java.lang.Thread.run(Thread.java:840)
```

Such issue may happen in Podfile parser if it contains lines with comments or something other than expacted:
```
# Uncomment the next line to define a global platform for your project
platform :ios, '12.0'

pod 'OneTrust-CMP-XCFramework', '202407.1.0.0'
```

### PR DESCRIPTION
**Pull request by ekharkunov**
Fixed problem in Podfile parsing.
Added tests for Podfile parsing.

Fixes #546

### CODE REVIEW

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 1]
```diff
@@ -41,7 +45,7 @@
 @Service
 public class CocoaPodsService {
 
-    private class MainPodfile {
+    static class MainPodfile {
```
Comment (ekharkunov) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:48]: Make it package private to access from tests.

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 2]
```diff
@@ -59,6 +63,10 @@ public String toString() {
         }
     }
 
+    static MainPodfile createMainPodfile() {
```
Comment (ekharkunov) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:66]: Specifically for testing.

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 3]
```diff
@@ -420,35 +428,9 @@ private MainPodfile createMainPodfile(List<File> podFiles, File jobDirectory, Fi
             jobEnvContext.get("env.MACOS_VERSION_MIN").toString());
         mainPodfile.platform = (platform.contains("ios") ? "ios" : "osx");
 
-        // Load all Podfiles
```
Comment (ekharkunov) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:423]: Move to separate function to make it testable.

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 4]
```diff
@@ -1265,7 +1247,7 @@ private InstalledPods installPods(File jobDir, File workingDir, File generatedDi
         if (!podFileLock.exists()) {
             throw new ExtenderException("Unable to find Podfile.lock in directory " + dir);
         }
-        LOGGER.info("Podfile.lock:\n{}", FileUtils.readFileToString(podFileLock));
+        LOGGER.info("Podfile.lock:\n{}", FileUtils.readFileToString(podFileLock, Charset.defaultCharset()));
```
Comment (ekharkunov) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:1250]: Mitigate deprecation warning.

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 5]
```diff
@@ -1420,9 +1402,47 @@ private String execCommand(String command, File cwd) throws ExtenderException {
...
+                if (line.startsWith("platform :")) {
+                    String version = line.replaceFirst("platform :ios|platform :osx", "").replace(",", "").replace("'", "").trim();
+                    if (!version.isEmpty() && (compareVersions(version, mainPodfile.platformMinVersion) > 0)) {
+                        mainPodfile.platformMinVersion = version;
+                    }
+                }
+                else {
+                    pods.add(line);
 
+                    Matcher matcher = podPattern.matcher(line);
```
Comment (ekharkunov) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:1436]: Actual fix.

File: server/src/main/java/com/defold/extender/services/CocoaPodsService.java [hunk 6]
```diff
@@ -1420,9 +1402,47 @@ private String execCommand(String command, File cwd) throws ExtenderException {
...
+            // Split each file into lines and go through them one by one
+            // Search for a Podfile platform and version configuration, examples:
+            //   platform :ios, '9.0'
+            //   platform :osx, '10.2'
+            // Get the version and figure out which is the highest version defined. This
+            // version will be used in the combined Podfile created by this function. 
+            // Treat everything else as pods
+            List<String> lines = Files.readAllLines(podFile.getAbsoluteFile().toPath());
+            for (String line : lines) {
+                if (line.isEmpty()) {
```
Comment (ekharkunov) [inline server/src/main/java/com/defold/extender/services/CocoaPodsService.java:1424]: Check empty line before any string search.

**Review by britzl — APPROVED** [review]
Excellent! Good that you added some tests. I've felt bad about adding a bunch of code not covered by unit tests! 👍 🦸

**Pull request merged**